modules/calib3d/src/stereobm.cpp

```diff
@@ -1008,7 +1009,10 @@ struct FindStereoCorrespInvoker : public ParallelLoopBody
 #if CV_SIMD128
         if( useSIMD && useShorts )
         {
-            findStereoCorrespondenceBM_SIMD( left_i, right_i, disp_i, cost_i, *state, ptr, row0, rows - row1 );
+            if( disp_i.type() == CV_16S)
+                findStereoCorrespondenceBM_SIMD<short>( left_i, right_i, disp_i, cost_i, *state, ptr, row0, rows - row1 );
+            else
+                findStereoCorrespondenceBM_SIMD<int>( left_i, right_i, disp_i, cost_i, *state, ptr, row0, rows - row1 );
```

Comment:
Please add some check into "else" branch:
- `CV_CheckTypeEQ(disp_i.type(), CV_32S, "Only 16S / 32S types are allowed for 'disparity' output")` or "Internal error" message
- or `CV_Assert()` if type check is performed before

Reply:
done, see L1009

Reply:
In fact, the type check is done in `FindStereoCorrespInvoker` constructor, no need to do it again

Reply:
Purpose of assertion here (and in general of assertions) is to avoid errors in the future during code refactoring.
This "if" not complete, it assumes two values only.

For example, you want to add CV_32F type, adapt check somewhere (~50 lines from here), update other places, but don't change this block.
How many time would you waste for debugging, because program will not crash (sizeof(int) == sizeof(float)), just produce many NaNs?